src/otx/core/data/module.py

```diff
@@ -217,6 +275,63 @@ def predict_dataloader(self) -> DataLoader:
             persistent_workers=config.num_workers > 0,
         )
 
+    def unlabeled_dataloader(self) -> dict[str, DataLoader] | None:
+        """Returns a dictionary of unlabeled dataloaders.
+
+        The method creates and returns dataloaders for unlabeled datasets based on the configuration settings.
+        If the data root is not specified in the configuration, it returns None.
+
+        Returns:
+            dict[str, DataLoader] | None: A dictionary containing unlabeled dataloaders, where the keys are the names of
+            the datasets and the values are the corresponding DataLoader objects.
+        """
+        config = self.config.unlabeled_subset
+        if self.config.unlabeled_subset.data_root is None:
+            return None
+
+        common_args = {
+            "batch_size": config.batch_size,
+            "num_workers": config.num_workers,
+            "pin_memory": True,
+            "persistent_workers": config.num_workers > 0,
+        }
+
+        dataloaders = {}
+        if isinstance(config.transforms, dict):
+            log.warning(f"Unlabeled dataset has multiple transforms : {list(config.transforms.keys())}")
+            common_args["worker_init_fn"] = lambda _: torch.manual_seed(0)  # type: ignore[assignment]
```

Comment:
Is it OK to hardcode the seed? Do we need to change worker_init_fn?

Reply:
What I am considering here is the same dataset, but applying several different transforms. At this time, the unlabeled dataloader should return an index of the same dataset in the batch. So that code was added. Is there perhaps a better way?